Dfc.Api.DiscoverSkillsAndCareers.Repositories/UserSessionRepository.cs

```diff
@@ -1,27 +1,53 @@
 ﻿using DFC.Api.DiscoverSkillsAndCareers.Models;
 using Microsoft.Azure.Documents;
 using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.Documents.Linq;
+using System;
 using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Linq.Expressions;
 using System.Threading.Tasks;
 
 namespace DFC.Api.DiscoverSkillsAndCareers.Repositories
 {
     [ExcludeFromCodeCoverage]
     public class UserSessionRepository : IUserSessionRepository
     {
-        private readonly CosmosDbConnection cosmosSettings;
+        private readonly CosmosDbConnection cosmosDbConnection;
         private readonly IDocumentClient client;
 
-        public UserSessionRepository(IDocumentClient client, CosmosDbConnection cosmosSettings)
+        public UserSessionRepository(IDocumentClient client, CosmosDbConnection cosmosDbConnection)
         {
-            this.cosmosSettings = cosmosSettings;
+            this.cosmosDbConnection = cosmosDbConnection;
             this.client = client;
         }
 
+        private Uri DocumentCollectionUri => UriFactory.CreateDocumentCollectionUri(cosmosDbConnection.DatabaseId, cosmosDbConnection.UserSessionCollectionId);
+
         public async Task CreateUserSession(UserSession userSession)
         {
-            var uri = UriFactory.CreateDocumentCollectionUri(cosmosSettings.DatabaseId, cosmosSettings.UserSessionCollectionId);
-            await client.CreateDocumentAsync(uri, userSession).ConfigureAwait(false);
+            await client.CreateDocumentAsync(DocumentCollectionUri, userSession).ConfigureAwait(false);
+        }
+
+        public async Task<UserSession> GetAsync(Expression<Func<UserSession, bool>> where)
+        {
+            var query = client.CreateDocumentQuery<UserSession>(DocumentCollectionUri, new FeedOptions { MaxItemCount = 1, EnableCrossPartitionQuery = true })
```

Comment:
Cross partition query is going to be slow isnt it?

Reply:
Amended as suggested